redis/cache/redispipebp/monitored.go

```diff
@@ -60,19 +61,46 @@ func (s MonitoredSync) Do(ctx context.Context, cmd string, args ...interface{})
 		s.Name+".do",
 		tracing.SpanTypeOption{Type: tracing.SpanTypeClient},
 	)
+	command := extractCommand(cmd)
+	active := redisprom.ActiveRequests.With(prometheus.Labels{
+		redisprom.ClientNameLabel: s.Name,
+		redisprom.CommandLabel:    command,
+		redisprom.DatabaseLabel:   "", // We don't have that info
+		redisprom.TypeLabel:       "", // We don't have that info
+		redisprom.DeploymentLabel: "", // We don't have that info
+	})
 	defer func(start time.Time) {
 		err := redis.AsError(result)
+		durationSeconds := time.Since(start).Seconds()
 		promHistogram.With(prometheus.Labels{
 			labelSlug:    s.Name,
-			labelCommand: extractCommand(cmd),
+			labelCommand: command,
 			labelSuccess: prometheusbp.BoolString(err == nil),
-		}).Observe(time.Since(start).Seconds())
+		}).Observe(durationSeconds)
+		active.Dec()
+		redisprom.LatencySeconds.With(prometheus.Labels{
+			redisprom.ClientNameLabel: s.Name,
+			redisprom.CommandLabel:    command,
+			redisprom.SuccessLabel:    prometheusbp.BoolString(err == nil),
+			redisprom.DatabaseLabel:   "", // We don't have that info
+			redisprom.TypeLabel:       "", // We don't have that info
+			redisprom.DeploymentLabel: "", // We don't have that info
+		}).Observe(durationSeconds)
+		redisprom.RequestsTotal.With(prometheus.Labels{
+			redisprom.ClientNameLabel: s.Name,
+			redisprom.CommandLabel:    command,
+			redisprom.SuccessLabel:    prometheusbp.BoolString(err == nil),
+			redisprom.DatabaseLabel:   "", // We don't have that info
+			redisprom.TypeLabel:       "", // We don't have that info
+			redisprom.DeploymentLabel: "", // We don't have that info
+		}).Inc()
 		span.FinishWithOptions(tracing.FinishOptions{
 			Ctx: ctx,
 			Err: err,
 		}.Convert())
 	}(time.Now())
 
+	active.Inc()
```

Comment:
🔕  very minor nit, but I tend to have the "do" go before the corresponding deferred "undo"... `lock(); defer unlock()`, and `inc(); defer dec()` -- so I don't think it matters much if you don't want to go through all of these files and move them, but I think it would be very slightly clearer if you did

```
active.inc()
defer func(start time.Time) {
  active.dec()
  // ...
}
```

Reply:
updated